src/cryptsetup/cryptsetup.c

```diff
@@ -235,9 +235,25 @@ static int parse_one_option(const char *option) {
                 arg_readonly = true;
         else if (streq(option, "verify"))
                 arg_verify = true;
-        else if (streq(option, "silent"))
-                arg_silent = true;
-        else if (STR_IN_SET(option, "allow-discards", "discard"))
+        else if ((val = startswith(option, "password-echo="))) {
+                if (arg_ask_password_flags) {
+                        log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                        "Duplicate password-echo= option \"%s\", ignoring.", val);
+                        return 0;
+                }
+
+                if (streq(val, "off"))
+                        arg_ask_password_flags = ASK_PASSWORD_SILENT;
+                else if (streq(val, "on"))
+                        arg_ask_password_flags = ASK_PASSWORD_ECHO;
```

Comment:
please use `parse_boolean()` on `val`. That's how we pretty universally parse boolean/on/off style values. 

as mentioned above, unset the other flags here. i.e.

```c
if (streq(val, "masked")) 
        arg_ask_password_flags &= ~(ASK_PASSWORD_ECHO|ASK_PASSWORD_SILENT);
else {
        r = parse_boolean(val);
        if (r < 0) {
                       log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Invalid password= option \"%s\", ignoring.", val);
                       return 0;
        }

        SET_FLAG(arg_ask_password_flags, ASK_PASSWORD_ECHO, b);
        SET_FLAG(arg_ask_password_flags, ASK_PASSWORD_SILENT, !b);
}
```